extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTableWriter.java

```diff
@@ -98,54 +98,70 @@ interface TransferObject<B> extends SafeCloseable {
+    // * @param incomingMeta A map of metadata values to be stores in the file footer
+    // * @param groupingPathFactory a factory to construct paths for grouping tables.
+    // * @param groupingColumns List of columns the tables are grouped by (the write operation will store the grouping
+    // * info)
+    // * @throws SchemaMappingException Error creating a parquet table schema for the given table (likely due to
+    // * unsupported types)
+    // * @throws IOException For file writing related errors
+    // */
+    // public static void write(
+    // @NotNull final Table t,
+    // @NotNull final String path,
+    // @NotNull final Map<String, String> incomingMeta,
+    // @NotNull final Function<String, String> groupingPathFactory,
+    // @NotNull final String... groupingColumns) throws SchemaMappingException, IOException {
+    // write(t, t.getDefinition(), ParquetInstructions.EMPTY, path, incomingMeta, groupingPathFactory,
+    // groupingColumns);
+    // }
+
+    // /**
+    // * Writes a table in parquet format under a given path
+    // *
+    // * @param t the table to write
+    // * @param path the destination path
+    // * @param incomingMeta any metadata to include in the parquet metadata
+    // * @param groupingColumns the grouping columns (if any)
+    // */
+    // public static void write(@NotNull final Table t,
+    // @NotNull final String path,
+    // @NotNull final Map<String, String> incomingMeta,
+    // @NotNull final String... groupingColumns) throws SchemaMappingException, IOException {
+    // write(t, path, incomingMeta, defaultGroupingFileName(path), groupingColumns);
+    // }
 
     /**
-     * Writes a table in parquet format under a given path
-     *
-     * @param t The table to write
-     * @param path The destination path
-     * @param incomingMeta A map of metadata values to be stores in the file footer
-     * @param groupingPathFactory a factory to construct paths for grouping tables.
-     * @param groupingColumns List of columns the tables are grouped by (the write operation will store the grouping
-     *        info)
-     * @throws SchemaMappingException Error creating a parquet table schema for the given table (likely due to
-     *         unsupported types)
-     * @throws IOException For file writing related errors
+     * Helper struct used to pass information about where to write the grouping files for each grouping column
      */
-    public static void write(
-            @NotNull final Table t,
-            @NotNull final String path,
-            @NotNull final Map<String, String> incomingMeta,
-            @NotNull final Function<String, String> groupingPathFactory,
-            @NotNull final String... groupingColumns) throws SchemaMappingException, IOException {
-        write(t, t.getDefinition(), ParquetInstructions.EMPTY, path, incomingMeta, groupingPathFactory,
-                groupingColumns);
-    }
+    static final class GroupingColumnWritingInfo {
+        /**
```

Comment:
We need to at least make the class and constructor public.
```
        /**
         * Parquet name of this grouping column
         */
        public final String parquetColumnName;
        /**
         * File path to be added in the grouping metadata of main parquet file
         */
        public final File metadataFilePath;

        /**
         * Destination path for writing the grouping file. The two filenames can differ because we write grouping files
         * to shadow file paths first and then place them at the final path once the write is complete. But the metadata
         * should always hold the accurate path.
         */
        public final File destFile;

        public GroupingColumnWritingInfo(final String parquetColumnName, final File metadataFilePath, final File destFile) {
            this.parquetColumnName = parquetColumnName;
            this.metadataFilePath = metadataFilePath;
            this.destFile = destFile;
        }
```